build-system/tasks/check-types.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const $$ = require('gulp-load-plugins')();
+const gulp = $$.help(require('gulp'));
+const log = require('fancy-log');
+const {cleanupBuildDir, closureCompile} = require('../compile/compile');
+const {compileCss} = require('./css');
+const {createCtrlcHandler, exitCtrlcHandler} = require('../ctrlcHandler');
+const {extensions, maybeInitializeExtensions} = require('./extension-helpers');
+const {isTravisBuild} = require('../travis');
+const {startNailgunServer, stopNailgunServer} = require('./nailgun');
+
+const maybeUpdatePackages = isTravisBuild() ? [] : ['update-packages'];
+
+// Also used in closure-compile.js
+const NAILGUN_PORT = '2114';
+
+/**
+ * Dedicated type check path.
+ * @return {!Promise}
+ */
+function checkTypes() {
```

Comment:
Nit: This could be converted to an `async` function.

Reply:
This (and more) coming up during the `gulp` v4 upgrade.